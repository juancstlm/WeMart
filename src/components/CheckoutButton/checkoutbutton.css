.wmrt-CheckoutButton {
  width: 100%;
  padding: 8px;
}
.wmrt-CheckoutButton-button {
  text-decoration: none;
  background-color: #d30707;
  border-radius: 4px;
}
.wmrt-CheckoutButton-button a:hover {
  text-decoration: none;
  color: white;
}

.wmrt-CheckoutButton-button a {
  text-decoration: none;
  height: 48px;
  text-align: center;
  padding: 10px 8px;
  display: block;
  position: relative;
}

.wmrt-CheckoutButton-button:hover {
    background-color: #ba0606;
}
.wmrt-CheckoutButton-totalContainer {
  position: absolute;
  right: 8px;
  top: 0;
  bottom: 0;
  display: flex;
  align-items: center;
}
.wmrt-CheckoutButton-total {
  background: linear-gradient(rgba(20, 20, 20, 0.2), rgba(20, 20, 20, 0.2));
  padding: 7px;
  border-radius: 4px;
}
.wmrt-CheckoutButton-text {
  font-size: 20px;
  text-align: center;
  font-weight: 600;
  color: white;
}
.wmrt-CheckoutButton-total-fullPrice{
    text-decoration: line-through;
    margin-right: 8px;
    color: rgb(255, 255, 255);;
    font-weight: 600;
    opacity: .8;
}
.wmrt-CheckoutButton-total-savingsPrice{
    color: white;
    font-weight: 600;
}
