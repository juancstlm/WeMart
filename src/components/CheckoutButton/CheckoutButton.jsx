import React from "react";
import PropTypes from "prop-types";
import "./checkoutbutton.css";

const CheckOutButton = ({ total, savingsTotal, onClick }) => {
  const formattedTotal = total.toFixed(2);
  const formattedSavingsTotal = savingsTotal.toFixed(2);

  return (
    <div className={"wmrt-CheckoutButton"}>
      <div className={"wmrt-CheckoutButton-button"}>
        <a onClick={onClick}>
          <div className={"wmrt-CheckoutButton-text"}>Checkout</div>
          <div className={"wmrt-CheckoutButton-totalContainer"}>
            <div className={"wmrt-CheckoutButton-total"}>
              <span
                className={"wmrt-CheckoutButton-total-fullPrice"}
              >{`$${formattedTotal}`}</span>
              <span
                className={"wmrt-CheckoutButton-total-savingsPrice"}
              >{`$${formattedSavingsTotal}`}</span>
            </div>
          </div>
        </a>
      </div>
    </div>
  );
};

export default CheckOutButton;

CheckOutButton.propTypes = {
  /* The order total */
  total: PropTypes.number,
  /* The order total with all savings applied */
  savingsTotal: PropTypes.number,
  /* Function to execute when button is pressed */
  onClick: PropTypes.func
};
