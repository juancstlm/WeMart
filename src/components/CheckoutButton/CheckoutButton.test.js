import React from "react";
import CheckoutButton from "./CheckoutButton";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer
    .create(<CheckoutButton total={19.0} savingsTotal={18.22} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("Button executes onClick function", () => {
  // Create the mock function
  const fn = jest.fn();
  // crate the component
  const component = renderer.create(<CheckoutButton onClick={fn} />);
  //Simulate button click
  const button = component.root.findByType("button");
  button.props.onClick();
  //Verify callback is invoked
  expect(fn.mock.calls.length).toBe(1);
});

test("Displays proper number formatting", () => {
  let input = 19.0000001;
  const tree = renderer.create(<CheckoutButton total={input} />).toJSON();
  expect(tree).toMatchSnapshot();
});
