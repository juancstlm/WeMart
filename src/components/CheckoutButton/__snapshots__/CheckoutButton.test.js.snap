// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Displays proper number formatting 1`] = `
<div
  className="wmrt-CheckoutButton"
>
  <div
    className="wmrt-CheckoutButton-button"
  >
    <button>
      <div
        className="wmrt-CheckoutButton-text"
      >
        Checkout
      </div>
      <div
        className="wmrt-CheckoutButton-totalContainer"
      >
        <div
          className="wmrt-CheckoutButton-total"
        >
          <span
            className="wmrt-CheckoutButton-total-fullPrice"
          >
            $19.00
          </span>
          <span
            className="wmrt-CheckoutButton-total-savingsPrice"
          >
            $0.00
          </span>
        </div>
      </div>
    </button>
  </div>
</div>
`;

exports[`renders correctly 1`] = `
<div
  className="wmrt-CheckoutButton"
>
  <div
    className="wmrt-CheckoutButton-button"
  >
    <button>
      <div
        className="wmrt-CheckoutButton-text"
      >
        Checkout
      </div>
      <div
        className="wmrt-CheckoutButton-totalContainer"
      >
        <div
          className="wmrt-CheckoutButton-total"
        >
          <span
            className="wmrt-CheckoutButton-total-fullPrice"
          >
            $19.00
          </span>
          <span
            className="wmrt-CheckoutButton-total-savingsPrice"
          >
            $18.22
          </span>
        </div>
      </div>
    </button>
  </div>
</div>
`;
